crates/kubelet/src/resources/quantity.rs

```diff
@@ -0,0 +1,338 @@
+//! A custom Quantity type that actually parses the `Quantity` type from k8s to something usable
+//! with real numbers
+
+// NOTE: This whole module might be useful to other people doing k8s things in Rust. We may want to
+// eventually export this or create a separate crate if people want to reuse it
+
+use k8s_openapi::apimachinery::pkg::api::resource::Quantity as KubeQuantity;
+
+const KILOBYTE: f64 = 1000.0;
```

Comment:
How come they need to be floating point values? Shouldn't these be of type `u32`?

I can understand the need for floating point values for `QuantityType::Cpu`. but shouldn't we use whole numbers for `QuantityType:Memory`?

Reply:
Welcome to the terrible world of `Quantity`! So according to the spec, you can technically make a memory request of `1.5Gi` which is a float. Basically we convert everything to a float to make the operations easier, hence why these values are floats as well

Reply:
Wonderful.